### ISSUE TITLE
Tilemaps no longer reflect changes via live updates

### ISSUE THREAD

**Issue by waldnercharles** [godotengine/godot#77328]
### Godot version

v4.0.3.stable.mono.official [5222a99f5]

### System information

Windows 10

### Issue description

When updating a Tilemap's cells via the editor, the changes are not reflected in the game. This worked fine in 3.5

### Steps to reproduce

1. Create a TileMap and Tileset with a tile you'd like to paint with. (I've been testing with the godot icon.)
2. Make sure "Synchronize Scene Changes" is turned on.
3. Launch your game, and paint some tiles in the TileMap.
4. Save if you want, but, it shouldn't matter. Godot 3.5 behavior didn't require a save for changes to appear.

### Minimal reproduction project

N/A

**Comment by KoBeWi** [comment]
Sounds like a recent regression. Pretty sure it was working in 4.0.2 or 4.0.1.

**Comment by waldnercharles** [comment]
> Sounds like a recent regression. Pretty sure it was working in 4.0.2 or 4.0.1.

I decided to test to see if it was working in 4.0.1 or 4.0.2, but, it seems broken in both those versions on Windows.

I've linked the project I'm using just in case I'm doing something wrong. The repro steps are simple but, I'm still pretty new to Godot so, you never know.

[Godot_TileMapTest.zip](https://github.com/godotengine/godot/files/11535916/Godot_TileMapTest.zip)
